lib/posix/posix.nim

```diff
@@ -663,6 +663,25 @@ proc waitid*(a1: cint, a2: Id, a3: var SigInfo, a4: cint): cint {.
 proc waitpid*(a1: Pid, a2: var cint, a3: cint): Pid {.
   importc, header: "<sys/wait.h>".}
 
+type Rusage* {.importc: "struct rusage", header: "<sys/resource.h>",
+               pure, final.} = object
+  ru_utime*, ru_stime*: Timeval                       # User and system time
+  ru_maxrss*, ru_ixrss*, ru_idrss*, ru_isrss*,        # memory sizes
+    ru_minflt*, ru_majflt*, ru_nswap*,                # paging activity
+    ru_inblock*, ru_oublock*, ru_msgsnd*, ru_msgrcv*, # IO activity
+    ru_nsignals*, ru_nvcsw*, ru_nivcsw*: clong        # switching activity
+
+proc wait4*(pid: Pid, status: ptr cint, options: cint, rusage: ptr Rusage): Pid
+  {.importc, header: "<sys/wait.h>", discardable.}
```

Comment:
Not discardable.

Reply:
For `wait4` it's true that there are more error cases and often you want the return value.  "Often but not always" - you may be invoking `wait4` for its behavior - to pause the caller until a child is done and you only have one child in question and so don't care about the return value.  Anyway, this case is much more debatable, so I just made this change and put a comment next to `getrusage` about failure.